tests/integration/cmd_dev_test.go

```diff
@@ -712,6 +712,216 @@ ComponentSettings:
+					})
+
+					It("should react on the Devfile modification", func() {
+						By("not warning users that odo dev needs to be restarted", func() {
+							warning := "Please restart 'odo dev'"
+							Expect(stdout).ShouldNot(ContainSubstring(warning))
+							Expect(stderr).ShouldNot(ContainSubstring(warning))
+						})
+						By("updating the pod", func() {
+							podName := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
+							bufferOutput := commonVar.CliRunner.Run("get", "pods", podName, "-o", "jsonpath='{.spec.containers[0].resources.requests.memory}'").Out.Contents()
+							output := string(bufferOutput)
+							Expect(output).To(ContainSubstring("1023Mi"))
+						})
+
+						By("exposing the endpoint", func() {
+							Eventually(func(g Gomega) {
+								url := fmt.Sprintf("http://%s", ports[ContainerPort])
+								Expect(url).To(ContainSubstring(LocalPort))
+								resp, err := http.Get(url)
+								g.Expect(err).ToNot(HaveOccurred())
+								defer resp.Body.Close()
+
+								body, _ := io.ReadAll(resp.Body)
+								for _, i := range []string{"Hello from Node.js Starter Application!"} {
+									g.Expect(string(body)).To(ContainSubstring(i))
+								}
+								g.Expect(err).ToNot(HaveOccurred())
+							}).WithPolling(1 * time.Second).WithTimeout(20 * time.Second).Should(Succeed())
+						})
+					})
+				})
+			})
+		})
+
+		When("devfile has multiple endpoints", func() {
+			var (
+				LocalPort1 = helper.GetRandomFreePort()
+				LocalPort2 = helper.GetRandomFreePort()
+				LocalPort3 = helper.GetRandomFreePort()
+			)
+			const (
+				// ContainerPort<N> are hard-coded from devfile-with-multiple-endpoints.yaml
+				// Note 1:	Debug endpoints will not be exposed for this instance, so we do not add custom mapping for them.
+				// Note 2: We add custom mapping for all the endpoints so that none of them are assigned random ports from the 20001-30001 range;
+				// Note 2(contd.): this is to avoid a race condition where a test running in parallel is also assigned similar ranged port the one here, and we fail to access either of them.
+				ContainerPort1 = "3000"
+				ContainerPort2 = "4567"
+				ContainerPort3 = "7890"
+			)
+			BeforeEach(func() {
+				helper.Cmd("odo", "set", "project", commonVar.Project).ShouldPass()
+				helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project-with-multiple-endpoints"), commonVar.Context)
+				helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-multiple-endpoints.yaml")).ShouldPass()
+			})
+
+			When("running odo dev", func() {
+				var devSession helper.DevSession
+				var ports map[string]string
+				BeforeEach(func() {
+					opts := []string{fmt.Sprintf("--port-forward=%s:%s", LocalPort1, ContainerPort1), fmt.Sprintf("--port-forward=%s:%s", LocalPort2, ContainerPort2), fmt.Sprintf("--port-forward=%s:%s", LocalPort3, ContainerPort3)}
```

Comment:
It could be interesting to test with 3 endpoints on Devfile and only 2 custom forwarded ports, to be sure the other one is forwarded from 20001.

Reply:
Yes, you are right. Thanks